### PR TITLE
[IMP][account_currency_tools] backport from v8 and adapt to work

### PR DESCRIPTION
**Pull request by JulioSerna**

### CODE REVIEW

File: account_currency_tools/wizard/exchange_realization.py [hunk 1]
```diff
@@ -0,0 +1,960 @@
...
+            INNER JOIN account_account AS aa ON aa.id = aml.account_id
+            INNER JOIN account_period AS ap ON ap.id = aml.period_id
+            INNER JOIN account_move AS am ON am.id = aml.move_id
+            WHERE
+                aa.type = '%(account_type)s' AND
+                aml.currency_id IS NOT NULL AND
+                aml.state <> 'draft' AND
+                am.state IN (%(states)s) AND
+                aml.company_id = %(company_id)d AND
+                aa.parent_left BETWEEN %(parent_left)d AND %(parent_right)d AND
```
Comment (JulioSerna) [inline account_currency_tools/wizard/exchange_realization.py]: @hbto 

This is the main change in v8 `aa.id` and v7 I use `aa.parent_left`

Comment (hbto) [inline account_currency_tools/wizard/exchange_realization.py]: @moylop260 is this change sane?
the one that @JulioSerna comments here

Comment (moylop260) [inline account_currency_tools/wizard/exchange_realization.py]: I saw fine this change...
Just fix sql injection

Comment (hbto) [inline account_currency_tools/wizard/exchange_realization.py]: @JulioSerna then after fixing sql injection offending code this PR can be merged

Regards.

**Comment by JulioSerna** [comment]
@moylop260  @hbto 
Sql injection fixed

**Comment by hbto** [comment]
+1

**Comment by moylop260** [comment]
- [x] Fix SQL Injection
- [x] Fix get child similar to osv https://github.com/odoo/odoo/blob/fd8570770ddc6546a07a1b9876acf58510d31679/openerp/osv/expression.py#L723

:+1:

**Pull request merged**